modules/core/02-client/keeper/events.go

```diff
@@ -92,15 +115,13 @@ func EmitSubmitMisbehaviourEvent(ctx sdk.Context, clientID string, clientState e
 	)
 }
 
-// EmitSubmitMisbehaviourEventOnUpdate emits a client misbehaviour event on a client update event
-func EmitSubmitMisbehaviourEventOnUpdate(ctx sdk.Context, clientID string, clientState exported.ClientState, consensusHeight exported.Height, headerStr string) {
-	ctx.EventManager().EmitEvent(
+// EmitUpgradeChainEvent emits an upgrade chain event.
+func EmitUpgradeChainEvent(ctx sdk.Context, height int64) {
+	ctx.EventManager().EmitEvents(sdk.Events{
 		sdk.NewEvent(
-			types.EventTypeSubmitMisbehaviour,
-			sdk.NewAttribute(types.AttributeKeyClientID, clientID),
-			sdk.NewAttribute(types.AttributeKeyClientType, clientState.ClientType()),
-			sdk.NewAttribute(types.AttributeKeyConsensusHeight, consensusHeight.String()),
-			sdk.NewAttribute(types.AttributeKeyHeader, headerStr),
+			types.EventTypeUpgradeChain,
```

Comment:
does the ConsensusHeight still need to be emitted here?

Reply:
No, not for `EmitSubmitMisbehaviourEventOnUpdate`. That information doesn't really help since there's no action to take when the client becomes frozen. The misbehaviour used to freeze the client can always be queried from the tx 

For `EmitUpgradeChainEvent` the important information is the plan height as that is where the consensus state is stored